### ISSUE TITLE
Rethink auto playing of sub-items

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#1004]
Currently, it is quite convenient (and probably the most likely use-case) that when a user plays e.g. a YouTube "watch" URL that they actually want the video to play.

Playing a watch URL actually generates sub-items and one of these sub-items contains the actual media streaming URL.

Other playlist files will work similarly, like cue sheets or playlist files.

vlcj now plays sub-items automatically by associating a native media list player instance with the native media player instance. This is all completely encapsulated and vlcj users never even know this is going on.

This is great, it works, and is almost certainly what most people will want to happen.

The downside is it takes some control away from the developer in that it is no longer optional to begin actual playback of the media.

Other use cases might be play a CD to get the list of sub-items, then allow the user to select which one to play, rather than playing the first immediately.

This can sort of be worked around by using mediaPlayer#startPaused(), even so, this is not an ideal situation.

So, need a rethink on how best to handle this.

It might be as simple as making the use of the media list player optional and enabled/disabled via an API call.

**Comment by caprica** [comment]
The use case in question was already possible, but it was a bit convoluted as per the Javadoc in SubItemApi#changeMedia (which is not actually public API):

```
     * If media was played, the sub-item list will start playing automatically. If you do <em>not</em> want this, then
     * instead of playing the media you should prepare it, then parse it, and wait for the parsed status changed media
     * event.
```

**Comment by caprica** [comment]
Resolved with b428af1aa07951382c5e261d9e1156f3e1c67fd9.

Quite a simple change really to make sub-item handling contingent on a new "mode", currently to either do nothing, or to play the first item.
